Fix freeing not allocated fallback entries by allocating a copy.

diff --git a/ncurses/tinfo/lib_setup.c b/ncurses/tinfo/lib_setup.c
--- a/ncurses/tinfo/lib_setup.c
+++ b/ncurses/tinfo/lib_setup.c
@@ -755,12 +755,12 @@ TINFO_SETUP_TERM(TERMINAL **tp,
 	/* try fallback list if entry on disk */
 	if (status != TGETENT_YES) {
 	    const TERMTYPE2 *fallback = _nc_fallback2(tname);
 
 	    if (fallback) {
 		T(("found fallback entry"));
-		TerminalType(termp) = *fallback;
+		_nc_copy_termtype2(&TerminalType(termp), fallback);
 		status = TGETENT_YES;
 	    }
 	}
 
 	if (status != TGETENT_YES) {
diff --git a/ncurses/tinfo/tinfo_driver.c b/ncurses/tinfo/tinfo_driver.c
--- a/ncurses/tinfo/tinfo_driver.c
+++ b/ncurses/tinfo/tinfo_driver.c
@@ -171,12 +171,12 @@ drv_CanHandle(TERMINAL_CONTROL_BLOCK * TCB, const char *tname, int *errret)
     /* try fallback list if entry on disk */
     if (status != TGETENT_YES) {
 	const TERMTYPE2 *fallback = _nc_fallback2(tname);
 
 	if (fallback) {
 	    T(("found fallback entry"));
-	    TerminalType(termp) = *fallback;
+	    _nc_copy_termtype2(&TerminalType(termp), fallback);
 	    status = TGETENT_YES;
 	}
     }
 
     if (status != TGETENT_YES) {
